pallets/pips/src/benchmarking.rs

```diff
@@ -0,0 +1,180 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(feature = "runtime-benchmarks")]
+use crate::*;
+use frame_benchmarking::benchmarks;
+use frame_support::assert_ok;
+use frame_system::RawOrigin;
+use pallet_identity::{self as identity, benchmarking::make_account};
+use polymesh_common_utilities::MaybeBlock;
+use sp_std::{
+    convert::{TryFrom, TryInto},
+    prelude::*,
+};
+
+benchmarks! {
+    _ {}
+
+    set_prune_historical_pips {
+        let origin = RawOrigin::Root;
+    }: _(origin, true)
```

Comment:
Nit: Maybe do `let b in 0..2;` + `let f = b as bool;` and use that as the argument to the extrinsic?

Reply:
Clippy complains about `as bool` and asks to write it as `b != 0`.

Reply:
`b != 0` also works fine; my main point is to exercise both `true` and `false`, not just `false`.

Reply:
The `..` notation seems to hide `..=`. When I tried `0 .. 2` before it used 2 as well!

Reply:
There is a related problem. With 200 samples I get 199 ones and only one zero. This is by no means a good distribution. I notice the same with other ranges: the top of the range (the `..top` - which means the range is inclusive despite the `..` notation`) is by far the most frequent sample value. I agree that sometimes biasing like that might make sense but it's doing it all the time.

Reply:
Maybe this `..` being inclusive is a substrate specific thing?

Reply:
Yes, this is where it's defined: https://github.com/paritytech/substrate/blob/095d0f06a42b66a92f28d097499a69e3f170d59d/frame/benchmarking/src/lib.rs#L901.

Reply:
Slightly tangential: The default value for `PruneHistoricalPips` is false so setting it to true will always be the worst case. If we want to test the false case as well, we first need to set it to true and then benchmark setting it to false, or else it becomes a no-op instead of the worst-case.

Considering all things, I'd suggest leaving this as it is.